File: py/solns/numDifferentIntsInStr/numDifferentIntsInStr.py

class Solution:
    @staticmethod
    def naive(word):
        count = []
        s = set("0123456789")
        prev = ""
        for tmps in word:
            if tmps in s:
                prev+=tmps
            else:
                if len(prev)!=0 and int(prev) not in count:
                    count.append(int(prev))
                prev = ""
        if len(prev)!=0 and int(prev) not in count:
            count.append(int(prev))
        return len(count)

File: cpp/example/src/NumDifferentIntsInStr/numDifferentIntsInStr.cpp
#include "numDifferentIntsInStr.h"

int numDifferentIntsInStr::naive(string word){
    set<int> count = {};
    set<int> s = {'1','2','3','4','5','6','7','8','9','0'};
    set<int>::iterator it = count.begin();
    string prev = "";
    for (char tmps : word){
        if (s.find(tmps)!=s.end()){
            prev += tmps;
        } else {
            if (prev.length()!=0 && 
                count.find(stoi(prev))== count.end()){
                    count.insert(it,stoi(prev));
                }
            prev = "";
        }
    }
    if (prev.length()!=0 && 
        count.find(stoi(prev))== count.end()){
            count.insert(it,stoi(prev));
        }
    return count.size();
}
